apps/simple_test_app.cc

```diff
@@ -0,0 +1,35 @@
+/**
+ * @file simple_test_app to show basic functionality of DAQProcess
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2020.
+ * Licensing/copyright details are in the COPYING file that you should have received with this code.
+ */
+
+#include "app-framework/DAQProcess.hh"
+#include "app-framework/DebugLoggingUserModule.hh"
+#include "app-framework/QueryResponseCommandFacility.hh"
+
+namespace appframework {
+std::unique_ptr<CommandFacility> CommandFacility::handle_ = std::unique_ptr<CommandFacility>(new QueryResponseCommandFacility());
+
+class simple_test_app_ModuleList : public ModuleList {
+    // Inherited via ModuleList
+    virtual void ConstructGraph(BufferMap& buffer_map, UserModuleMap& user_module_map, TransitionOrderMap& transition_order_map) override {
+        user_module_map["debugLogger"].reset(new DebugLoggingUserModule());
+    }
+};
+}  // namespace appframework
+
+int main(int argc, char* argv[]) {
+    std::list<std::string> args;
+    for (int ii = 1; ii < argc; ++ii) {
+        args.push_back(std::string(argv[ii]));
+    }
+
+    appframework::DAQProcess theDAQProcess(args);
+
+    auto ml = std::unique_ptr<appframework::ModuleList>(new appframework::simple_test_app_ModuleList());
+    theDAQProcess.register_modules(ml);
+
```

Comment:
It would be good to package this even further.  It seems `ml` is wholly internal and nothing that the `main()` cares about.  Further factoring would allow multiple `DAQProcess` instances to run in separate threads of a given process.

One major benefit of the actor pattern is to have the flexibility to aggregate actors in processes or actors as processes.

Reply:
Could multiple DAQProcesses instances in a given process talk to each other (presumably via the same types of Buffers/Queues that UserModules within a DAQProcess can talk to each other)?  I suspect yes...